src/main/java/software/amazon/event/kafkaconnector/offloading/S3EventBridgeEventDetailValueOffloading.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.offloading;
+
+import static com.fasterxml.jackson.core.JsonPointer.compile;
+import static java.util.stream.Collectors.partitioningBy;
+import static java.util.stream.Collectors.toList;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.Error.reportOnly;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.failure;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.success;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.event.kafkaconnector.EventBridgeResult;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class S3EventBridgeEventDetailValueOffloading
+    implements EventBridgeEventDetailValueOffloadingStrategy {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(S3EventBridgeEventDetailValueOffloading.class);
+
+  private final S3Client client;
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  public S3EventBridgeEventDetailValueOffloading(final S3Client client) {
```

Comment:
question: I'm wondering if these classes should get the JSON field path to offload as a parameter/configuration?

Reply:
It's WIP, the parameter will be added as it goes on. BTW Jackson supports [JSON Pointer ](https://datatracker.ietf.org/doc/html/rfc6901) and not JSON Path. Instead of `$.detail.value[.]...` it will be `/detail/value[/]...`.

Reply:
We might need to discuss this with @baldawar because I feel the ergonomics are going to suffer from different path spec expressions.

A simple way would be to map `$.` syntax to `/` but it has edges which we need to guard against :/

Reply:
[JSON Path](https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html) is not yet a RFC. From the Draft:
> JSONPath is not intended as a replacement for, but as a more powerful companion to, JSON Pointer [[RFC6901](https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html#RFC6901)].

With JSON Path it's possible to do 'path traversal', that means the validation is more complex. Nevertheless the Java library (Jayway) [JsonPath](https://github.com/json-path/JsonPath) can be configured to use Jackson via SPI. 

Currently, the code make use of the implementation of Jackson's JSON Path that returns a pointer to the node and can be mutated easily.

Reply:
> Using https://github.com/json-path/JsonPath.

SGTM!